File: app/src/main/java/com/yhjoo/dochef/model/MultiItemTheme.java
package com.yhjoo.dochef.model;

import com.chad.library.adapter.base.entity.MultiItemEntity;

public class MultiItemTheme implements MultiItemEntity {
    int itemType;
    int spanSize;
    Recipe content;

    public MultiItemTheme(int itemType, int spanSize, Recipe content) {
        this.itemType = itemType;
        this.spanSize = spanSize;
        this.content = content;
    }

    public MultiItemTheme(int itemType, int spanSize) {
        this.itemType = itemType;
        this.spanSize = spanSize;
    }

    public int getSpanSize() {
        return spanSize;
    }

    public Recipe getContent() {
        return content;
    }

    @Override
    public int getItemType() {
        return itemType;
    }
}

File: app/src/main/java/com/yhjoo/dochef/model/MultiItemTheme.kt
package com.yhjoo.dochef.model

import com.chad.library.adapter.base.entity.MultiItemEntity

class MultiItemTheme : MultiItemEntity {
    private var multiType: Int
    var spanSize: Int
    var content: Recipe? = null

    constructor(itemType: Int, spanSize: Int, content: Recipe?) {
        this.multiType = itemType
        this.spanSize = spanSize
        this.content = content
    }

    constructor(itemType: Int, spanSize: Int) {
        this.multiType = itemType
        this.spanSize = spanSize
    }

    override fun getItemType(): Int {
        return multiType
    }
}